test/integration/net_test.go

```diff
@@ -32,6 +32,60 @@ import (
 	"k8s.io/minikube/pkg/util/retry"
 )
 
+// TestCNIFalse checks that minikube returns and error
+// if container runtime is "containerd" or "crio"
+// and --cni=false
+func TestCNIFalse(t *testing.T) {
+
+	for _, cr := range []string{"containerd", "cri-o"} {
+		t.Run("TestCNIFalse-"+cr, func(t *testing.T) {
+			profile := UniqueProfileName("no-cni-" + cr)
+			ctx, cancel := context.WithTimeout(context.Background(), Minutes(1))
+			defer CleanupWithLogs(t, profile, cancel)
+
+			startArgs := []string{"start", "-p", profile, "--memory=2048", "--alsologtostderr", "--cni=false", "--container-runtime=" + cr}
+			startArgs = append(startArgs, StartArgs()...)
+			mkCmd := exec.CommandContext(ctx, Target(), startArgs...)
+			rr, err := Run(t, mkCmd)
+			if err == nil {
+				t.Errorf("%s expected to fail", mkCmd)
+				continue
+			}
+			expectedMsg := fmt.Sprintf("The %q container runtime requires CNI", cr)
+			if !strings.Contains(rr.Output(), expectedMsg) {
+				t.Errorf("Expected %q line not found in ouput %s", expectedMsg, rr.Output())
+			}
+		})
+	}
+}
+
+// TestCNIFalseForce checks that minikube returns not error
+// if container runtime is "containerd" or "crio"
+// and --cni=false, but --force=true
+func TestCNIFalseForce(t *testing.T) {
```

Comment:
instead of creating a new test, lets add this as part of the same SubTest that we have `--cni=false`
```
			{"false", []string{"--cni=false"}, "", "", false},

```



in which we only expect exit Usage code if --cni=false and container runtime is not Docker and in case of docker it continue same test as before

```
if strings.Contains(tc.name, "false") { 
..... if Non Docker ...
          Expect Exit Usage
If Docker (continue the test as it is...now)


}



```

Reply:
Not sure, actually. `TestNetworkPlugins` is focused on how network is configured.  In`TestCNIFalseForce` we mostly check how the command line arguments are validated. These tests check different parts of minikube code. Also,`TestNetworkPlugins` is already too big, overloaded with checks and hard to understand. Adding extra 20 lines wouldn't make it better.

Reply:
--force is an option we discourage users form using, and it means you are at your own risk and not supported, we do not have tests for --force and I am certainly okay for skipping testing if --force gives a warning message to users if it comes at the cost of starting a new minikube. our integration test machines are limitted specially on expensive options like hyperkit and mac and we can not add a new minikube start for this option.

lets move it under Cni tests as a subtest